lib/pure/random.nim

```diff
@@ -309,7 +309,7 @@ proc rand*[T: Ordinal or SomeFloat](r: var Rand; x: HSlice[T, T]): T =
     doAssert r.rand(1..6) == 4
     doAssert r.rand(1..6) == 6
     let f = r.rand(-1.0 .. 1.0) # 0.8741183448756229
-
+  assert x.a <= x.b, "Start of slice must be less than end."
```

Comment:
`"Start of slice must be less than end."` is redundant with the error message assert already provides:
```
Error: unhandled exception: random.nim(312, 10) `x.a <= x.b` Start of slice must be less than end. [AssertionDefect]
```

I'd either leave that message out (just `assert x.a <= x.b`) or show runtime values, eg `assert x.a <= x.b, $(x.a, x.b)` or with a formatted msg (see `chks.nim`)

Reply:
Disagree, it's not clear at all what `x` refers to or `a` for many who aren't familiar with slices in Nim.

Runtime values would be nice, but tbh I would argue `assert` should have some magic to print them.

Reply:
>  I would argue assert should have some magic to print them.

this I agree with; this magic (or rather macro) could be enabled by default in debug builds, to expand out the runtime values of lhs and rhs in an infix expression, similar to unittest.check, so long it doesn't affect semantics. Definitely doable.